### PR TITLE
Add new parameter for gen_pyi.py to make it more configureable.

### PR DESCRIPTION
**Pull request by qlzh727**
This will make the script more flexible for the directory where it is executed.

### CODE REVIEW

**Comment by qlzh727** [comment]
@kiukchung for visibility.

File: tools/autograd/build.bzl [hunk 1]
```diff
@@ -12,3 +12,9 @@ def define_targets(rules):
             "//torchgen",
         ],
     )
+
+    rules.filegroup(
+        name = "deprecated_yaml",
+        srcs = ["deprecated.yaml"],
+        visibility = ["//:__subpackages__"],
+    )
```
Comment (soulitzer) [inline tools/autograd/build.bzl:20]: Why is this needed?

**Review by soulitzer — COMMENTED** [review]
Added a small question, otherwise SGTM

File: tools/autograd/build.bzl (hunk 1 above)
Comment (qlzh727) [reply tools/autograd/build.bzl:20]: Thanks for the question. We plan to use the deprecated_yaml from a blaze genrule that invokes pyi.py. As the input to the pyi.py, genrule requires the input file to be explicitly listed out. When we feed the value of tools/autograd/deprecated.yaml to genrule, it failed to resolve since tools/autograd is a package from blaze perspective. Any file under a blaze package will a proper blaze target to be access.

Comment (kiukchung) [reply tools/autograd/build.bzl:20]: `deprecated.yaml` is one of the arguments (`--deprecated-functions-path`) to `tools/pyi/gen_pyi.py`. This PR doesn't include the genrule for generating the `pyi` files (aka actually calling `gen_pyi.py`) though.

@qlzh727 as discussed offline, we should include the genrule `:generate_pyi_files` in this PR so that the PR is actually context-complete.

Lets add (to `<root>/build.bzl`)

1. the `py_binary` target for `tools/pyi/gen_pyi.py`
	```
	py_binary(
	    name = "gen_pyi",
	    srcs = [
	        "tools/pyi/gen_pyi.py",
	    ],
	    main = "tools/pyi/gen_pyi.py",
	    deps = [
	        "@pytorch//:tools/autograd",
	        "@pytorch//:tools/torchgen",
	    ],
	)
	```

2. the `genrule` to call `:gen_pyi`:
	```
	genrule(
	    name = "generate_pyi_files",
	    srcs = [
	        "aten/src/ATen/native/native_functions.yaml",
	        "aten/src/ATen/native/tags.yaml",
	        # The torch/tools/autograd is a subpackage, we have to explicitly list it as target here.
	        "@pytorch//:tools/autograd:deprecated_yaml",
	        # pyi.in inputs
	        "torch/_C/return_types.pyi.in",
	        "torch/_C/_VariableFunctions.pyi.in",
	        "torch/_C/__init__.pyi.in",
	        "torch/_C/_nn.pyi.in",
	        "torch/nn/functional.pyi.in",
	    ],
	    outs = [
	        "torch/_VF.pyi",
	        "torch/return_types.pyi",
	        "torch/_C/_VariableFunctions.pyi",
	        "torch/_C/__init__.pyi",
	        "torch/_C/_nn.pyi",
	        "torch/nn/functional.pyi",
	    ],
	    cmd = """
	    $(location :gen_pyi) \
	        --native-functions-path $(location aten/src/ATen/native/native_functions.yaml) \
	        --tags-path $(location aten/src/ATen/native/tags.yaml) \
	        --deprecated-functions-path $(location @pytorch//:tools/autograd:deprecated_yaml) \
	        --out `dirname $(@D)` \
	        --template-dir `dirname $(location torch/_C/return_types.pyi.in)`/../..
	    """,
	    tools = [":gen_pyi"],
	)
	```
